File: frontend/apps/web/lib/roles.ts
export const ROLES = {
  GiamDoc: "GiamDoc",
  DoiTruong: "DoiTruong",
  NhanVien: "NhanVien",
} as const;

export type Role = typeof ROLES[keyof typeof ROLES];

export const ROLE_LABELS: Record<string, string> = {
  [ROLES.GiamDoc]: "Giám Đốc",
  [ROLES.DoiTruong]: "Đội Trưởng",
  [ROLES.NhanVien]: "Nhân Viên",
};

export function getRoleLabel(role?: string | null): string {
  if (!role) return "";
  return ROLE_LABELS[role] || role;
}

File: backend/src/Domain/Constants/Roles.cs
namespace backend.Domain.Constants;

/// <summary>
/// Hệ thống 3 vai trò chính trong Quản Lý Cây Xanh.
/// </summary>
public abstract class Roles
{
    /// <summary>Giám Đốc — phê duyệt kế hoạch, quyết định cấp cao.</summary>
    public const string GiamDoc = nameof(GiamDoc);

    /// <summary>Đội Trưởng Cây Xanh — quản trị hệ thống, quản lý công việc (admin-like).</summary>
    public const string DoiTruong = nameof(DoiTruong);

    /// <summary>Nhân Viên Cây — thực hiện công việc thực địa.</summary>
    public const string NhanVien = nameof(NhanVien);
}